### PR TITLE
feat(ui): add custom source form to images page

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added custom source form to images page

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/images and check that changing the source type radio to "Custom" shows the custom source form
- Enter a random url and submit the form - check that an error shows above the form
- Click "Show advanced" and enter some values, then click "Hide advanced" then "Show advanced" again. Check that the advanced values were cleared.
- Enter a valid URL (e.g. https://images.maas.io/ephemeral-v3/stable/) and check that images are successfully fetched and show up in `fetchedImages` in state

## Fixes

Fixes canonical-web-and-design/app-squad#86

## Screenshot
Screenshot 2021-06-11 at 16-47-47 Images bolla MAAS

### CODE REVIEW

File: ui/src/app/store/bootresource/slice.ts [hunk 1]
```diff
@@ -46,7 +46,9 @@ const bootResourceSlice = createSlice({
           model: BootResourceMeta.MODEL,
           method: "delete_image",
         },
-        payload: params,
+        payload: {
+          params,
+        },
```
Comment (Caleb-Ellis) [inline ui/src/app/store/bootresource/slice.ts:51]: accidentally set all of these wrong, lolz

Comment (huwshimi) [reply ui/src/app/store/bootresource/slice.ts:51]: I've done that before.

File: ui/src/app/images/views/ImageList/UbuntuImages/CustomSourceForm/CustomSourceForm.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+const CustomSourceSchema = Yup.object()
+  .shape({
+    keyring_data: Yup.string(),
+    keyring_filename: Yup.string(),
+    url: Yup.string().required("URL is required"),
+  })
+  .defined();
+
+export type CustomSourceValues = {
```
Comment (huwshimi) [inline ui/src/app/images/views/ImageList/UbuntuImages/CustomSourceForm/CustomSourceForm.tsx:24]: Should this be associated with the `BootResourceUbuntuSource` type at all?

**Review by huwshimi — APPROVED** [review]
Nice work! One small comment.

**Pull request merged**